File: src/main/java/visnode/pdi/process/BrightnessProcess.java
package visnode.pdi.process;

import org.paim.commons.Image;
import org.paim.commons.ImageFactory;
import visnode.commons.Input;
import visnode.commons.Output;
import visnode.pdi.Process;

/**
 * Process for adjusting the brightness of the image
 */
public class BrightnessProcess implements Process {
    
    /** Brightness process*/
    private final org.paim.pdi.BrightnessProcess process;

    /**
     * Creates a new brightness process
     *
     * @param image
     * @param brightness
     */
    public BrightnessProcess(@Input("image") Image image, @Input("brightness") Integer brightness) {
        Image resultImage = image;
        if (image == null) {
            resultImage = ImageFactory.buildEmptyImage();
        }
        this.process = new org.paim.pdi.BrightnessProcess(new Image(resultImage), brightness);
    }

    @Override
    public void process() {
        process.process();
    }

    /**
     * Returns the output image
     * 
     * @return Image
     */
    @Output("image")
    public Image getImage() {
        return process.getOutput();
    }
    
}

File: src/main/resources/visnode/pdi/process/BrightnessProcess.js
function process(image) {
    var resultImage = ImageFactory.buildEmptyImage(image)
    var brightness = 90
    for (var c = 0; c < image.getChannelCount(); c++) {
        for (var x = 0; x < image.getWidth(); x++) {
            for (var y = 0; y < image.getHeight(); y++) {
                // Calculate the brightness value
                var newPixel = image.get(c, x, y) + brightness;
                resultImage.set(c, x, y, resultImage.getPixelValueRange().limit(newPixel))
            }
        }
    }
    return resultImage
}